app/settings/user-settings/user-settings.controller.js

```diff
@@ -34,29 +52,52 @@ function (
 
     UserSettingsEndpoint.getFresh({id: $rootScope.currentUser.userId}).$promise.then((settings) => {
         _.each(settings.results, (setting) => {
+            setting.user_id = setting.user.id;
+
             if (setting.config_key === 'hdx_api_key') {
-                $scope.hxlApiKey = '*** *** *** *** *** *** *** ' + setting.config_value.slice(setting.config_value.length - 4);
-                $scope.hxlApiId = setting.id;
+                setting.config_value = '*** *** *** *** *** *** *** ' + setting.config_value.slice(setting.config_value.length - 4);
+                $scope.hdxSettings.hdx_api_key = setting;
+                $scope.hxlApiKeySet = true;
+            }
+            if (setting.config_key === 'hdx_maintainer_id') {
+                $scope.hxlMaintainerSet = true;
+                $scope.hdxSettings.hdx_user_id = setting;
             }
         });
     });
 
-    function newKey() {
-        $scope.hxlApiKey = false;
+    function changeKey() {
+        $scope.hxlApiKeySet = false;
+    }
+
+    function changeId() {
+        $scope.hxlMaintainerSet = false;
     }
 
     function goToHdxView() {
         $state.go('settings.hdx');
     }
 
     function saveKey() {
-        $scope.api.$dirty = true;
-        if ($scope.apiKey) {
-            UserSettingsEndpoint.saveCache({id: $scope.hxlApiId,  user_id: $rootScope.currentUser.userId,  config_key: 'hdx_api_key', config_value: $scope.apiKey}).$promise.then((response) => {
-                $scope.hxlApiKey = '*** *** *** *** *** *** *** ' + $scope.apiKey.slice($scope.apiKey.length - 4);
-                $scope.hxlApiId = response.id;
-                Notify.notifyAction('settings.user_settings.api_key_saved', null, false, 'thumb-up', 'circle-icon confirmation', {callback: goToHdxView, text: 'settings.user_settings.start_tagging', callbackArg: null, actionClass: 'button button-alpha'});
-            });
+        var calls = [];
```

Comment:
I think I found a bug. 
Prerequisite
- [ ] having the API key and maintainer id set up
Steps
- [ ]  change your maintainer id, save
    - [ ] you will receive a validation error but it won't show up (only in network/console). Example : 
{"errors":[{"status":422,"message":"Failed to validate user_settings entity:\nThe user id, 2, and config key, hdx_user_id, already exist"},{"status":422,"title":"The user id, 2, and config key, hdx_user_id, already exist","message":"The user id, 2, and config key, hdx_user_id, already exist","source":{"pointer":"\/config_key"}}]}